hyperspy/_components/exponential.py

```diff
@@ -57,3 +61,97 @@ def __init__(self, A=1., tau=1., module="numexpr", **kwargs):
         )
 
         self.isbackground = False
+
+    def estimate_parameters(self, signal, x1, x2, only_current=False):
+        """Estimate the parameters for the exponential component by splitting
+        the signal window into two regions and using their geometric means
+
+        Parameters
+        ----------
+        signal : BaseSignal instance
+        x1 : float
+            Defines the left limit of the spectral range to use for the
+            estimation.
+        x2 : float
+            Defines the right limit of the spectral range to use for the
+            estimation.
+
+        only_current : bool
+            If False estimates the parameters for the full dataset.
+
+        Returns
+        -------
+        bool
+
+        """
+        super(Exponential, self)._estimate_parameters(signal)
+        axis = signal.axes_manager.signal_axes[0]
+        i1, i2 = axis.value_range_to_indices(x1, x2)
+        if i1 + 1 == i2:
+            if i2 < axis.high_index:
+                i2 += 1
+            elif i1 > axis.low_index:
+                i1 -= 1
+        i_mid = (i1 + i2) // 2
+        x_start = axis.index2value(i1)
+        x_mid = axis.index2value(i_mid)
+        x_end = axis.index2value(i2)
+
+        if only_current is True:
+            s = signal.get_current_signal()
+        else:
+            s = signal
+
+        if s._lazy:
+            import dask.array as da
+            exp = da.exp
+            log = da.log
+        else:
+            exp = np.exp
+            log = np.log
+
+        with np.errstate(divide='raise'):
+            try:
+                # use log and exp to compute geometric mean to avoid overflow
+                a1 = s.isig[i1:i_mid].data
+                b1 = log(a1)
+                a2 = s.isig[i_mid:i2].data
+                b2 = log(a2)
+                geo_mean1 = exp(b1.sum(axis=-1)/b1.shape[-1])
+                geo_mean2 = exp(b2.sum(axis=-1)/b2.shape[-1])
+                x1 = (x_start + x_mid) / 2
+                x2 = (x_mid + x_end) / 2
+
+                A = exp((log(geo_mean1) - (x1 / x2) * log(geo_mean2)) /
+                        (1 - x1 / x2))
+                t = -x2 / (log(geo_mean2) - log(A))
+
+                if s._lazy:
+                    A = A.map_blocks(np.nan_to_num)
+                    t = t.map_blocks(np.nan_to_num)
+                else:
+                    A = np.nan_to_num(A)
+                    t = np.nan_to_num(t)
+
+            except (FloatingPointError):
```

Comment:
When testing, I had the following warning:
```python
/home/eric/Dev/hyperspy/hyperspy/_components/exponential.py:120: RuntimeWarning: invalid value encountered in double_scalars
  geo_mean1 = exp(b1.sum(axis=-1)/b1.shape[-1])
/home/eric/Dev/hyperspy/hyperspy/_components/exponential.py:121: RuntimeWarning: invalid value encountered in double_scalars
  geo_mean2 = exp(b2.sum(axis=-1)/b2.shape[-1])
```
I think the `RuntimeWarning` could be added too here.

Reply:
I've added RuntimeWarning to the `except` block, but I'm curious what test caused the warning.  Everything I saw while working on it was some variation of divide-by-zero.

Reply:
I was testing it interactively and it happens most likely because of the signal_range being too small. To reproduce:
```python
import hyperspy.api as hs

exponential = hs.model.components1D.Exponential()
exponential.A.value = 12500.
exponential.tau.value = 168.

signal = hs.signals.Signal1D(
    exponential.function(np.arange(100, 200, 0.02)))

# use a signal range too small
signal.remove_background(signal_range=(19.425, 19.426),
            background_type='Exponential',
            show_progressbar=None)
```

Reply:
Thanks, that was helpful!  I've changed it to raise an exception if the range is too small and then catch it and log an appropriate warning.